design/one-pager-stacks-versioning.md

```diff
@@ -0,0 +1,248 @@
+    * Tags are a symbolic reference to an actual sha of an image and they can be updated or deleted
+ * I want to have context from a version:
+    * A sha is hard to compare when debugging
+      * Which is newer? - cb44dd0 or d2fb617?
+      * Is this alpha, beta?
+    * A tag like latest is opaque to the end user and so not incredibly useful:
+      * Which version am I running? you need to run special commands to figure out which sha is loaded
+      * on a node and then lookup that sha to see which version it's tagged against, if it's even still tagged.
+      * If the latest tag on this node the same as the latest tag on that node? When did it get restarted?
+
+As a note, if a user wants to continue to install from an arbitrary tag, stack-installer will work as expected, but
+ behavior relying on re-tagging latest and the stack installer detecting the tags underlying container being updated
+ will not trigger a re-install as the Crossplane stack installer by convention will assume that docker tags are
+ immutable and represent a single version. Any system that relies on tagging all builds with a :latest tag for example
+ is not recommended and in practice won't work as expected.
+
+#### Why standardize on semantic versions?
+
+Semantic versions are a well adopted standard that give you context just from looking at the version. Based on semantic
+ versions we could implement a variety of useful features, that we couldn't reasonably or reliably do if everyone is
+ using a unique versioning system.
+
+* Auto-upgrade a major, minor, patch version based on policy.
+* Detect if an upgrade is available based only on the tag and the latest version in a channel, no introspecting of a
+  docker package.
+* Warning if the next latest version is a major upgrade.
+* Compare versions running easily based on status of stack install.
+* Tie human readable versions to change logs.
+* I can remember a version, it's very hard to remember a sha.
+* Semantic versions used by helm and widely adopted in kubernetes community.
+
+## Design
+
+We will rely on the established standard of [semantic version 2.0](https://semver.org/spec/v2.0.0.html) embedding these
+ versions inside of the stack app.yaml metadata, and inside of the tag in a docker image.
+
+The stack app.yaml version should always be a simple stable semantic version with no pre-release metadata, and tags
+ must exist that match the semantic version in the release channel.
+
+Creation dates will be tracked in the release channel versions for auditing purposes and will be iso-8601 format.
+ This is the same format as kubernetes and is a well established standard. We will also include a sha id of the
+ underlying stack container, which is immutable, and will be used by the installer to fetch the exact artifact
+ referenced in the release channel.
+
+The stack install CRD will get a new optional field for channel in the spec, that will include the url to the channel.
```

Comment:
What is a release channel? What's the mechanism the design proposes for implementing a Thing Which Is A Release Channel? I am noticing some references to release channel URLs and I see the description of an s3 bucket later in the document, but I am imagining it may make sense to create a `### Release Channels` section which explains the structure of a channel and its API contract, to help put these other descriptions in context.